### PR TITLE
Add support for job configuration as a dictionary

### PR DESCRIPTION
**Pull request by clebergnu**
Which is part of the "Job API enabler" work, while keeping
compatibility with previous Avocado version that supply an
argparse.Namespace instance.

Reference: https://github.com/avocado-framework/avocado/pull/3248
Signed-off-by: Cleber Rosa <crosa@redhat.com>

### CODE REVIEW

File: avocado_vt/loader.py [hunk 1]
```diff
@@ -108,15 +109,19 @@ def __init__(self, args, extra_params):
...
             else:
-                self.args.vt_extra_params = vt_extra_params
+                extra = vt_extra_params
+            set_opt(self.args, 'vt_extra_params', extra)
 
     def _fill_optional_args(self):
         def _add_if_not_exist(arg, value):
-            if not hasattr(self.args, arg):
-                setattr(self.args, arg, value)
+            #if not hasattr(self.args, arg):
```
Comment (wainersm) [inline avocado_vt/loader.py]: delete left comment

Comment (clebergnu) [reply avocado_vt/loader.py]: Oops!

**Comment by wainersm** [comment]
I did bootstrap avocado-vt and run the unattended install test this PR and with latest Avocado (included https://github.com/avocado-framework/avocado/pull/3248). It seems all working.

1) Bootstrap:
`# avocado vt-bootstrap --vt-type qemu`

2) Ran unattended install:
```
avocado run unattended_install.url.extra_cdrom_ks.default_install.aio_native --vt-type qemu --vt-guest-os Linux.RHEL.8.devel.x86_64 --vt-machine-type q35 --vt-extra-params url=http://download.devel.redhat.com/released/RHEL-8/8.0.0/BaseOS/x86_64/os/ nettype=bridge kernel_params='ks=cdrom nic delay=60 console=ttyS0,115200 console=tty0 inst.repo=http://download.devel.redhat.com/released/RHEL-8/8.0.0/BaseOS/x86_64/os/'
```

File: avocado_vt/loader.py (hunk 1 above)
Comment (clebergnu) [reply avocado_vt/loader.py]: Fixed!

File: virttest/bootstrap.py [hunk 2]
```diff
@@ -902,46 +905,47 @@ def bootstrap(options, interactive=False):
...
     sync_download_dir(interactive)
 
-    test_dir = data_dir.get_backend_dir(options.vt_type)
-    if options.vt_type == 'libvirt':
+    test_dir = data_dir.get_backend_dir(vt_type)
+    if vt_type == 'libvirt':
         step = create_config_files(test_dir, shared_dir, interactive,
-                                   options.vt_type, step,
+                                   vt_type, step,
                                    force_update=options.vt_update_config)
```
Comment (luckyh) [inline virttest/bootstrap.py]: Just found this line has to be updated as well.

**Comment by clebergnu** [comment]
@ldoktor you say Travis is broken... care to show where?

**Comment by ldoktor** [comment]
Hello @clebergnu, here: https://travis-ci.org/avocado-framework/avocado-vt/jobs/570794803 you get a false-negative caused by this change. Yes, it is marked as allow_failure, but people still might be confused about the failure and might spend time reproducing it on their laptops.

File: avocado_vt/plugins/vt_bootstrap.py [hunk 3]
```diff
@@ -96,14 +96,14 @@ def configure(self, parser):
                             help=("The architecture of the distro to be used when "
                                   "generating the host configuration entry."))
 
-    def run(self, args):
+    def run(self, config):
```
Comment (balamuruhans) [inline avocado_vt/plugins/vt_bootstrap.py]: we need to use `args` itself else rebase the PR with commit 39f4ba79 and change the below lines to `config`,
https://github.com/avocado-framework/avocado-vt/blob/1efb1c3722918ef517670c4a518c03dfde250d9f/avocado_vt/plugins/vt_bootstrap.py#L107-L108

It breaks the framework with,
```
Avocado crashed:
Traceback (most recent call last):
  File "/usr/local/bin/avocado", line 4, in <module>
    __import__('pkg_resources').run_script('avocado-framework==70.0', 'avocado')
  File "/usr/local/lib/python3.7/site-packages/pkg_resources/__init__.py", line 666, in run_script
    self.require(requires)[0].run_script(script_name, ns)
  File "/usr/local/lib/python3.7/site-packages/pkg_resources/__init__.py", line 1462, in run_script
    exec(code, namespace, namespace)
  File "/usr/local/lib/python3.7/site-packages/avocado_framework-70.0-py3.7.egg/EGG-INFO/scripts/avocado", line 78, in <module>
    sys.exit(app.run())
  File "/usr/local/lib/python3.7/site-packages/avocado_framework-70.0-py3.7.egg/avocado/core/app.py", line 91, in run
    return method(self.parser.config)
  File "/usr/local/lib/python3.7/site-packages/avocado_framework_plugins_vt-70.0-py3.7.egg/avocado_vt/plugins/vt_bootstrap.py", line 107, in run
    if isinstance(self.args, dict):
AttributeError: 'VTBootstrap' object has no attribute 'args'
```

**Comment by ldoktor** [comment]
@clebergnu do you intend to work on this or should it be closed (and forgotten until something happens?)

**Comment by clebergnu** [comment]
@ldoktor I do intend to send a new version here.  I'm waiting on #2253 so that I'll be able to cover all Avocado version combinations here.

**Comment by clebergnu** [comment]
@ldoktor @balamuruhans @sathnaga @luckyh this is a very late, but still needed update on this PR.  The Job API is really coming on Avocado now, and we need to be prepared here on Avocado-VT.  Now with the CI also testing both qemu and libvirt bootstraps, this will hopefully catch all cases.  Still, I'd appreciate your reviews and testing.  Thanks!

**Comment by clebergnu** [comment]
We have two reviews, and I did some extended testing with Avocado master ( https://paste.centos.org/view/4092ed6e ):

```
JOB ID     : c196e6d9b360b6e5a17fd05cf953d622a47a57d7
JOB LOG    : /home/cleber/avocado/job-results/job-2020-06-30T13.55-c196e6d/job.log
 (1/1) io-github-autotest-qemu.boot: PASS (20.38 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/cleber/avocado/job-results/job-2020-06-30T13.55-c196e6d/results.html
JOB TIME   : 28.11 s
```

Avocado 80.0 ( https://paste.centos.org/view/1473ad17 ):

```
JOB ID     : fdb6a1bc9ad28bbd53cbad06132bc33df5dc6fcf
JOB LOG    : /home/cleber/avocado/job-results/job-2020-06-30T13.57-fdb6a1b/job.log
 (1/1) io-github-autotest-qemu.boot: PASS (19.72 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/cleber/avocado/job-results/job-2020-06-30T13.57-fdb6a1b/results.html
JOB TIME   : 27.97 s
```

And Avocado 69.2 ( https://paste.centos.org/view/648ceb98 ):

```
JOB ID     : 22a286fa6db773a8e244f33e634380cf21066916
JOB LOG    : /home/cleber/avocado/job-results/job-2020-06-30T13.59-22a286f/job.log
 (1/1) io-github-autotest-qemu.boot: PASS (19.97 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 27.35 s
JOB HTML   : /home/cleber/avocado/job-results/job-2020-06-30T13.59-22a286f/results.html
```

**Pull request merged**